core/src/main/scala/org/http4s/headers/Keep-Alive.scala

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2013 http4s.org
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.http4s
+package headers
+
+import cats.implicits._
+import cats.parse._
+import org.http4s.internal.parsing.Rfc7230
+import org.http4s.util.{Renderable, Writer}
+import org.http4s.Header
+import org.typelevel.ci._
+import scala.concurrent.duration.FiniteDuration
+import java.util.concurrent.TimeUnit
+import scala.collection.mutable.ListBuffer
+import cats.kernel.Semigroup
+
+object `Keep-Alive` {
+
+  sealed trait KeepAlive
+  final case class Timeout(timeoutSeconds: Long) extends KeepAlive
+  final case class Max(max: Long) extends KeepAlive
+  final case class Extension(ext: (String, Option[String])) extends KeepAlive
+
+  /*
+https://www.w3.org/Protocols/HTTP/1.1/draft-ietf-http-v11-spec-01#Keep-Alive
+
+Keep-Alive           = "Keep-Alive" ":" 1#keep-alive-info
+keep-alive-info      =   "timeout" "=" delta-seconds
+                       / "max" "=" 1*DIGIT
+                       / keep-alive-extension
+keep-alive-extension = token [ "=" ( token / quoted-string ) ]
+   */
+
+  def apply(
+      timeoutSeconds: Option[Long],
+      max: Option[Long],
+      extension: List[(String, Option[String])]): ParseResult[`Keep-Alive`] =
```

Comment:
It is [an error for MIME types](https://datatracker.ietf.org/doc/html/rfc6838#page-13) to repeat parameters.  Parameters must be unique in their context in [structured fields](https://datatracker.ietf.org/doc/html/rfc8941#section-3.1.2).  I can't find anything in the `Cache-Control` spec.  I can't think of any other header whose parameters take this form that speciflcally allow duplicates, but our specification didn't _disallow_ it.

The spec is also silent on whether the keys are case sensitive or not.  

I think we did it reasonably, but we're left to guess on all of this. :weary:

Reply:
What if I'm a big jerk and call this `Keep-Alive(None, None, List("timeout", "30".some))`?

Reply:
@rossabaker Think just a heads up on the decisions we've made in a comment is good on this?  

Maybe you shouldn't be a big jerk ;)

Reply:
Well, without some effort, it won't round trip.  In addition to the handwaving about how extension keys shouldn't be non-tokens, they shouldn't be the two known tokens.  We can fix it in this constructor, but it's a headache.

I think we have a similar problem with CacheDirectives today: I don't remember what prevents a custom one from duplicating a modeled one.

Reply:
That's a good point that the round trip ends up with different case class representations.  😵‍💫 .

Reply:
I've fixed this up to not allow the reserved tokens in the extension list in the apply with a applicable error.  I did not do anything with the mime types repeat params, case sensitivity etc.